front/src/Phaser/Game/GameScene.ts

```diff
@@ -432,7 +432,7 @@ export class GameScene extends DirtyScene {
 
         gameManager.gameSceneIsCreated(this);
         urlManager.pushRoomIdToUrl(this.room);
-        analyticsClient.enteredRoom(this.room.id);
+        analyticsClient.enteredRoom(this.room.id, this.room.getInstance());
```

Comment:
We should not merge this. This will cause troubles down the road.

`getInstance` should really be deprecated because it enforces a URL pattern on the front (the idea is that the front should be agnostic with URL patterns and let those be decided by the pusher or the admin)

Code for getInstance is now:

```
    public getInstance(): string {
        if (this.instance !== undefined) {
            return this.instance;
        }

        if (this.isPublic) {
            const match = /_\/([^/]+)\/.+/.exec(this.id);
            if (!match) throw new Error('Could not extract instance from "' + this.id + '"');
            this.instance = match[1];
            return this.instance;
        } else {
            const match = /@\/([^/]+)\/([^/]+)\/.+/.exec(this.id);
            if (!match) throw new Error('Could not extract instance from "' + this.id + '"');
            this.instance = match[1] + "/" + match[2];
            return this.instance;
        }
    }

```

This means that is a URL dos not start with _ or @, it will trigger an exception. But people developing other admins might want to do URL starting with something else.

Reply:
I'm not sure what we should put instead to be honest. Maybe the admin/pusher could send a "group/world" information to the front that the front could send back to analytics

Reply:
well I am plannig to add a notion of room group soon, that could be used to describe worlds. Once this is done i could add them in the analytics